content/projects/gsoc/2019/plugin-installation-manager-tool-cli.adoc

```diff
@@ -26,25 +26,76 @@ There are currently many reincarnations of plugin management across Jenkins, inc
-Phase 2 will focus on improving upon what was written in Phase 1 and full integration with link:https://github.com/jenkinsci/docker[Jenkins Docker] and the link:https://github.com/jenkinsci/configuration-as-code-plugin[Configuration as Code Plugin].
-More robust input parsing and support to view security warnings and available updates for plugins will be included in these improvements.
+The first alpha version of the library and CLI tool was released. It implemented basic functionality
+of the install-plugins.sh script to take in plugins and download them to a user-specified directory.
+
+The alpha release was announced in the following link:/blog/2019/07/02/plugin-management-tool-alpha-release/[blog post]
+and a live demo can be seen in link:https://youtu.be/MDs0Vr7gnnA?t=196[the Phase 1 evaluation recording]
+(link:https://docs.google.com/presentation/d/12Bo8w9SinrG5n82w-Unjx4MNq0mjkHFEOMc3Jf6rTQQ/edit#slide=id.p1[slides])
+
+=== Phase 2
+Phase 2 focused on improving upon what was written in Phase 1, with particular focus on making parsing more robust,
+adding better Docker compatibility, and creating new CLI options that will make the plugin installation process
+more transparent. Parsing was improved to filter out comments and blank lines, and accept a yaml file listing plugins to install
+ as input. Instead of hard coding the update centers, options were added to allow update centers to be set via environment variables or CLI options.
+Windows compatible defaults for locations to find an existing Jenkins war file and setting the plugin directory were also added.
+To make the plugin installation process more transparent, options were added to view the list of plugins that would be installed prior
+to installation, and view if any of these plugins had available updates or security warnings.  This last step involved re-writing
+some of the plugin download and dependency resolution logic so that all plugins and their recursive dependencies are resolved
+before the time of download.
+
+The blog post announcing the new features can be found link:/blog/2019/07/30/plugin-management-tool-phase2-updates/[here] and the demo
+for this coding phase can be found link:https://www.youtube.com/watch?v=HlENuZZq7zc&feature=youtu.be&t=221[here]
+(link:https://docs.google.com/presentation/d/1wsM09IGgoA7gSsA9tW5fHlFAhl6B9fXlr-4iEP89SCk/edit?usp=sharing[slides])
+
+=== Phase 3
+Most of this phase was dedicated to cleaning up and testing the work done in Phase 1 and 2. Some fixes and optimizations included in
+this coding phase include throwing an exception sooner if there are problems downloading a plugin or resolving its dependencies,
+throwing an exception if the required Jenkins version listed for any of the plugins to be installed is lower than the version
+of the Jenkins war file, and improving plugin download times through a custom IO thread pool and using http HEAD instead of http GET where appropriate.
```

Comment:
Random: after updating this to include the link to Phase 3 demos after release, I think you could mention that this is in incrementals if people want to try out the bleeding edge.